### ISSUE TITLE
The focus should be visible on the homepage (WCAG 2.1 SC 2.4.7)

### ISSUE THREAD

**Issue by rpkoller** [mastodon/mastodon#19928]
### Steps to reproduce the problem

1. go to your homepage (in my case https://mastodon.social/home)
2. with the theme set to mastodon dark
3. tab through the interface

### Expected behaviour

an element should be visible in focus on each and every tab

### Actual behaviour

on a few tabs the focus has a too low color contrast on other tabs no focus is visible at all

### Detailed description

I've checked if the focus is visible on the main page of my account on https://mastodon.social/home . In general there are two or three tabs on every toot card without a visible focus outline. Ideally the user should be able to tab through the page and on each tab there should be a new visible focus outline.  The already visible focus states have also a few issues: 

### Search field - magnifying glass outline
![Screen Shot 2022-11-06 at 18 26 42](https://user-images.githubusercontent.com/1665422/200194700-ac189b3e-e95c-4c7c-bd02-9f28076d32e7.jpg)

Outline: #3a5999
Background: #3b3e48
Color contrast: 1.58:1 (should have at least 3:1)

### All links in the left sidebar (screenshot is just one example)
![Screen Shot 2022-11-06 at 18 27 04](https://user-images.githubusercontent.com/1665422/200194750-f0deea92-350a-4a22-988b-ba178ca2c513.jpg)

Outline: #325291
Background: #2b2d33
Color contrast: 1.8:1  (should have at least 3:1)

### Posts-, Hashtags-, News-tabs
![Screen Shot 2022-11-06 at 18 08 53](https://user-images.githubusercontent.com/1665422/200195002-40e848ae-7e76-4418-ae5f-fa0cfa2e4f86.jpg)

Outline: #355595
Background(left sidebar): #2c2d32 1.71:1 (should have at least 3:1)
Background (explore bar above): #434753. 1.23:1 (should have at least 3:1)
Background (tab nav) #32353c 1.69:1 (should have at least 3:1)
Background (under tab nav)  #3b3e48 1.47:1 (should have at least 3:1) 

_*there is no outline on all of the four sides and as noted the backgrounds aren' consistent on the outside and therefor different color contrasts for each to me the 3:1 color contrast_

### Close button
![Screen Shot 2022-11-06 at 18 11 16](https://user-images.githubusercontent.com/1665422/200195019-316ede49-ac79-49d7-9f43-33bcf9a4f9a7.jpg)

Outline: #4b505e
Background: #3b3e48
Color contrast: 1.32:1 (should have at least 3:1)

_*ideally i would use the grey background the button has on focus on hover instead and use the focus used on the rest of the page for the close button as well_

### Whole toot focus
![Screen Shot 2022-11-06 at 18 15 15](https://user-images.githubusercontent.com/1665422/200195039-aa10b7b1-dc7b-4151-9d3d-3b21d787afcb.jpg)

Outline: #3a5a9b 
Background (toot card) #3b3e48 1.58:1  (should have at least 3:1)
Background (sidebar) #2b2d33 2.04:1  (should have at least 3:1)

_* the outline is only visible at the bottom -  left, right and top are not visible_

### Within toot 
![Screen Shot 2022-11-06 at 18 32 46](https://user-images.githubusercontent.com/1665422/200195056-0281b037-f94b-44dd-9bbc-152c3de302aa.jpg)

Outline: #3a5a9b
Background #3b3e48 
Color contrast: 1.56:1 (should have at least 3:1)

### Toot action icons 
![Screen Shot 2022-11-06 at 22 05 34](https://user-images.githubusercontent.com/1665422/200195117-2a3d3e7c-1dd1-4a2e-abf0-7ae92fd94207.jpg)

Outline: #4b505e
Background: #3b3e48
Color Contrast: 1.32:1 (should have at least 3:1) 
_*ideally i would use the grey background the button has on focus on hover instead and use the focus used on the rest of the page for the close button as well_

### Right sidebar. focus for column links (unregistered user)
![Screen Shot 2022-11-06 at 18 41 30](https://user-images.githubusercontent.com/1665422/200195202-805b4b6c-c1d8-4112-b5ac-30fd671bb3a9.jpg)

_*they have no outline and the text color changes in slight white nuances or maybe the font weight changes as well? hard to distinguish any change_

### Right sidebar. focus for column links (registered user - advanced view)
![Screen Shot 2022-11-06 at 18 43 00](https://user-images.githubusercontent.com/1665422/200195347-8564599a-dc6b-4e8f-8b7c-b1f6cf7571aa.jpg)

Outline: #525766
Background: #4b505e
Color Contrast 1.12:1 (should have at least 3:1)

_*ideally i would add the focus outline used on the rest of the page for the column links as well_

### Sign in button
![Screen Shot 2022-11-06 at 22 11 25](https://user-images.githubusercontent.com/1665422/200195402-65018f96-47cd-485f-91ab-2fffe5794adc.jpg)

Outline: #32508e
Background (left sidebar): #272d33.   1.77:1 (should have at least 3:1)  
Bckground (sign in button): #7375f7.  2.1:1 (should have at least 3:1) 

### Create account. 
![Screen Shot 2022-11-06 at 22 20 28](https://user-images.githubusercontent.com/1665422/200195799-d0d96682-9d29-4f1c-adb6-74c1b812786b.jpg)

_*changes from a ghost button to a primary button but without an outline._ 

![Screen Shot 2022-11-06 at 22 21 05](https://user-images.githubusercontent.com/1665422/200195815-e2858151-512f-45f4-9427-f2aeab511095.jpg)
_*in contrast the learn more button in the left sidebar isn't becoming primary button when in focus_ 

### Specifications

Mastodon v.4.0.0rc1 using mastodon dark (on https://mastodon.social/home)

Tested with Safari 16.1 and Edge 107.0.1418.35 on MacOS Monterey

**Comment by diondiondion** [comment]
Thank you for the detailed report! This should be vastly improved in Mastodon 4.6, so I will close this issue now. If you come across any remaining focus visibility problems in Mastodon 4.6, please open new issues for those.
